core/src/main/java/org/mule/serialization/internal/ClassSpecificObjectSerializer.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.serialization.internal;
+
+import static java.lang.String.format;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.serialization.ObjectSerializer;
+import org.mule.util.ObjectInputStreamProvider;
+import org.mule.util.SerializationUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InvalidClassException;
+import java.io.ObjectInputStream;
+import java.io.ObjectStreamClass;
+import java.io.StreamCorruptedException;
+
+import org.apache.commons.io.input.ClassLoaderObjectInputStream;
+
+public class ClassSpecificObjectSerializer extends JavaObjectSerializer
+{
+    private Class classToAccept;
+
+    public ClassSpecificObjectSerializer(ObjectSerializer baseSerializer, Class classToAccept)
+    {
+        super(baseSerializer);
+        this.classToAccept = classToAccept;
+    }
+
+    @Override
+    protected <T> T doDeserialize(InputStream inputStream, ClassLoader classLoader) throws Exception
+    {
+        checkArgument(inputStream != null, "Cannot deserialize a null stream");
+        checkArgument(classLoader != null, "Cannot deserialize with a null classloader");
+
+        return (T) SerializationUtils.deserialize(inputStream, classLoader, muleContext, new ObjectInputStreamProvider()
```

Comment:
static import.

Reply:
The base class (AbstractObjectSerializer) already declares a deserialize() method

Reply:
You're right.